integrationTests/vm/staking/stakingV4_test.go

```diff
@@ -0,0 +1,221 @@
+package staking
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func requireSliceContains(t *testing.T, s1, s2 [][]byte) {
+	for _, elemInS2 := range s2 {
+		require.Contains(t, s1, elemInS2)
+	}
+}
+
+func requireSliceContainsNumOfElements(t *testing.T, s1, s2 [][]byte, numOfElements int) {
+	foundCt := 0
+	for _, elemInS2 := range s2 {
+		if searchInSlice(s1, elemInS2) {
+			foundCt++
+		}
+	}
+
+	require.Equal(t, numOfElements, foundCt)
+}
+
+func requireSameSliceDifferentOrder(t *testing.T, s1, s2 [][]byte) {
+	require.Equal(t, len(s1), len(s2))
+
+	for _, elemInS1 := range s1 {
+		require.Contains(t, s2, elemInS1)
+	}
+}
+
+func searchInSlice(s1 [][]byte, s2 []byte) bool {
+	for _, elemInS1 := range s1 {
+		if bytes.Equal(elemInS1, s2) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func searchInMap(validatorMap map[uint32][][]byte, pk []byte) bool {
+	for _, validatorsInShard := range validatorMap {
+		for _, val := range validatorsInShard {
+			if bytes.Equal(val, pk) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func requireMapContains(t *testing.T, m map[uint32][][]byte, s [][]byte) {
+	for _, elemInSlice := range s {
+		require.True(t, searchInMap(m, elemInSlice))
+	}
+}
+
+func requireMapDoesNotContain(t *testing.T, m map[uint32][][]byte, s [][]byte) {
+	for _, elemInSlice := range s {
+		require.False(t, searchInMap(m, elemInSlice))
+	}
+}
+
+// TODO: Staking v4: more tests to check exactly which nodes have been selected/unselected from previous nodes config auction
```

Comment:
also try with different configurations of previous nodes config (before activation)

Reply:
Right, a different custom setup should be done here